ndcube/ndcube.py

```diff
@@ -366,6 +372,98 @@ def extra_coords(self):
                     "value": self._extra_coords_wcs_axis[key]["value"]}
         return result
 
+    def all_world_coords(self, axes=None):
+        """
+        Returns WCS coordinate values of all pixels for all axes.
+
+        Parameters
+        ----------
+        axes: `int` or `str`, or iterable of `int` or `str`
+            Axis number in numpy ordering or unique substring of world_axis_physical_types
+            of axes for which real world coordinates are desired.
+            Default=None implies all axes will be returned.
+
+        Returns
+        -------
+        axes_coords: `list` of `astropy.units.Quantity`
+            Real world coords for axes in order requested by user.
+
+        Example
+        -------
+        >>> NDCube.all_world_coords(('lat', 'lon')) # doctest: +SKIP
+        >>> NDCube.all_world_coords(2) # doctest: +SKIP
+
+        """
+        # Define the dimensions of the cube and the total number of axes.
+        cube_dimensions = np.array(self.dimensions.value, dtype=int)
+        n_dimensions = len(cube_dimensions)
+        world_axis_types = self.world_axis_physical_types
+        # Determine axis numbers of user supplied axes.
+        if axes is None:
+            int_axes = np.arange(n_dimensions)
+        else:
+            if isinstance(axes, int):
+                int_axes = np.array([axes])
+            elif isinstance(axes, str):
+                int_axes = np.array([
+                    utils.cube.get_axis_number_from_axis_name(axes, world_axis_types)])
+            else:
+                int_axes = []
+                for axis in axes:
+                    if isinstance(axis, int):
+                        int_axes.append(axis)
+                    elif isinstance(axis, str):
+                        int_axes.append(
+                            utils.cube.get_axis_number_from_axis_name(axis, world_axis_types))
+                int_axes = np.array(int_axes)
+        # Ensure user has not entered the same axis twice.
+        if len(np.unique(int_axes)) != len(int_axes):
+            raise ValueError("Same axis entered more than once.")
+        n_axes = len(int_axes)
+        axes_coords = [None] * n_axes
+        axes_translated = np.array([False] * n_axes)
+        # Determine which axes are dependent on others.
+        # Ensure the axes are in numerical order.
+        dependent_axes = np.empty(n_axes, dtype=object)
```

Comment:
I think you can replace this whole block with:
```python
wcs_get_dependent = partial(utils.wcs.get_dependent_axes, self.wcs)
dependent_axes = list(map(tuple, map(sorted, map(wcs_get_dependent, int_axes))))
```

:laughing: 

Having said that, I feel like you could just modifiy `utils.wcs.get_dependent_axes` to do a `sorted` in the right places.

Reply:
Ha!  Yeh, I actually forgot I'm not importing ```get_dependent_axes```, so I know it wont change unless we change it.  And currently it already returns the axes ordered.  So I've rewritten that code chunk to be two list comprehensions :D